tests/run/type_inference.pyx

```diff
@@ -907,3 +907,39 @@ def variable_with_name_of_type():
     assert typeof(str) == 'str object', typeof(str)
 
     return list, str
+
+def variable_with_name_of_type_2():
+    """
+    >>> variable_with_name_of_type_2()
+    ([], 'abc')
+    """
+    # Names like 'list.append' refer to the type and must be inferred as such,
+    # but a simple variable called 'list' is not the same and used to break type inference.
+    # See https://github.com/cython/cython/issues/6835
+    rest_list = []
+    list = []  # note: same name as type of value
+    list = list + rest_list
```

Comment:
I think you can just add this line to the test above, after the `+=` line. It doesn't seem worth a separate test function. Then the name distinction is `_builtin` vs. `_exttype`.